src/classes/class-maintenancemode.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Handles maintenance page for the store.
+ *
+ * @category   Plugins
+ * @package    WordPress
+ * @subpackage woocart-defaults
+ * @since      3.17.0
+ */
+
+namespace Niteo\WooCart\Defaults {
+
+	/**
+	 * Class MaintenanceMode
+	 *
+	 * @package Niteo\WooCart\Defaults
+	 */
+	class MaintenanceMode {
+
+		public function __construct() {
+			add_action( 'template_redirect', array( $this, 'maintenance_mode' ) );
+		}
+
+		/**
+		 * Perform checks to determine if the maintenance page needs to be
+		 * rendered instead of regular store.
+		 */
+		public function maintenance_mode() {
+			// Check option to determine the status
+			if ( get_option( 'woocart_maintenance_mode', false ) ) {
+				// Address of the current page
+				$url = 'https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+
+				if ( ! is_admin() ) {
+					/**
+					 * We are checking for admin role and WP pages to bypass.
+					 */
+
+					// WooCommerce Forgot Password
```

Comment:
What would be the case here?

Reply:
This is for the admin as well, very rare edge case.

WC changes the default `Forgot Password` to `my-account/forgot-password`